deployment/pypi/setup.py

```diff
@@ -15,6 +17,35 @@
 else:
     raise NotImplementedError('current platform {} not supported'.format(os_type))
 
+class AutoCompletion(install):
+    def run(self):
+        RELEASE_VERSION = os.popen('git describe --tags --abbrev=0').read().strip()
+        COMP_URL = 'https://raw.githubusercontent.com/microsoft/nni/{}/tools/bash-completion'.format(RELEASE_VERSION)
+        if os_type == 'Linux':
+            HOME = os.environ.get('HOME')
+            if not HOME:
+                install.run(self)
```

Comment:
what is `install.run(self)` used for here?

Reply:
Not quiet sure, seems that this line actually do the install action. Every example about how to use cmdclass has this line, like this [link](https://stackoverflow.com/questions/21915469/python-setuptools-install-requires-is-ignored-when-overriding-cmdclass)